presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2182,6 +2194,64 @@ private void analyzeOrderBy(Query node, Scope orderByScope)
             return analyzeOrderBy(node, sortItems, orderByScope);
         }
 
+        /**
+         * @return true if the Query / QuerySpecification containing the analyzed
+         * Limit or FetchFirst, must contain orderBy (i.e., for FetchFirst with ties).
+         */
+        private boolean analyzeLimit(Node node)
+        {
+            checkState(
+                    node instanceof FetchFirst || node instanceof Limit,
+                    "Invalid limit node type. Expected: FetchFirst or Limit. Actual: %s", node.getClass().getName());
+            if (node instanceof FetchFirst) {
+                return analyzeLimit((FetchFirst) node);
+            }
+            else {
+                return analyzeLimit((Limit) node);
+            }
+        }
+
+        private boolean analyzeLimit(FetchFirst node)
+        {
+            if (!node.getRowCount().isPresent()) {
+                analysis.setLimit(node, 1);
```

Comment:
When does this happen?

Reply:
`FETCH FIRST ROWS` without specifying actual number of rows is allowed, meaning the literal for fetch first quantity can be omitted, in which case it defaults to 1